### PR TITLE
Fixed the loading of ScriptProperty types that were serialized using an older typename in JSON Serialization files.

### LINKED ISSUE #14910: Scripting: Editor crashes on opening a level with a AzFramework::ScriptPropertyGenericClass present
**Describe the bug**
Editor crashes on opening a level with a AzFramework::ScriptPropertyGenericClass present.

**Steps to reproduce**
1. Launch the Editor.
2. Open the Levels\Prefab\Lua_Spawnables_DespawnOnEntityDeactivate level.

**Expected behavior**
Level opens without issue.

**Actual behavior**
Editor crashes:

```
Call Stack Trace:
25) 00007FFAE6A40207 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFAE9115530 (ntdll) : memset
23) 00007FFAE90FC876 (ntdll) : _C_specific_handler
22) 00007FFAE911241F (ntdll) : _chkstk
21) 00007FFAE90C14A4 (ntdll) : RtlRaiseException
20) 00007FFAE9110F4E (ntdll) : KiUserExceptionDispatcher
19) D:\GitHub\o3de\Code\Framework\AzFramework\AzFramework\Script\ScriptComponent.cpp (899) : AzFramework::ScriptPropertyGroup::GetProperty
18) D:\GitHub\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\ToolsComponents\ScriptEditorComponent.cpp (492) : AzToolsFramework::Components::ScriptEditorComponent::LoadProperties
17) D:\GitHub\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\ToolsComponents\ScriptEditorComponent.cpp (827) : AzToolsFramework::Components::ScriptEditorComponent::LoadProperties
16) D:\GitHub\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\ToolsComponents\ScriptEditorComponent.cpp (780) : AzToolsFramework::Components::ScriptEditorComponent::LoadScript
15) D:\GitHub\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\ToolsComponents\ScriptEditorComponent.cpp (955) : AzToolsFramework::Components::ScriptEditorComponent::OnAssetReloaded
14) D:\GitHub\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (146) : AZ::Internal::EBusContainer<AZ::Data::AssetEvents,AZ::Data::AssetEvents,1,1>::Dispatcher<AZ::EBus<AZ::Data::AssetEvents,AZ::Data::AssetEvents> >::Event<void (__cdecl AZ::Dat
13) D:\GitHub\o3de\Code\Framework\AzCore\AzCore\Asset\AssetManager.cpp (1851) : AZ::Data::AssetManager::NotifyAssetReady
12) D:\GitHub\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (148) : AZStd::Internal::function_util::get_invoker<void __cdecl(void),AZStd::Internal::function_util::function_obj_tag,AZStd::stateless_allocator>::call<`AZ::BusInternal::EBusB
11) D:\GitHub\o3de\Code\Framework\AzCore\AzCore\EBus\Policies.h (267) : AZ::EBusQueuePolicy<1,AZ::EBus<AZ::Data::AssetEvents,AZ::Data::AssetEvents>,AZStd::recursive_mutex>::Execute
10) D:\GitHub\o3de\Code\Framework\AzCore\AzCore\Asset\AssetManager.cpp (591) : AZ::Data::AssetManager::DispatchEvents
 9) D:\GitHub\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AZ::SystemTickEvents,AZ::SystemTickEvents,0,1>::Dispatcher<AZ::EBus<AZ::SystemTickEvents,AZ::SystemTickEvents> >::Broadcast<void (__cdecl AZ::Sy
 8) D:\GitHub\o3de\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1506) : AZ::ComponentApplication::TickSystem
 7) D:\GitHub\o3de\Code\Editor\CryEdit.cpp (2345) : CCryEditApp::IdleProcessing
 6) D:\GitHub\o3de\Code\Editor\CryEdit.cpp (2255) : CCryEditApp::OnIdle
 5) D:\GitHub\o3de\Code\Editor\Core\QtEditorApplication.cpp (320) : Editor::EditorQtApplication::maybeProcessIdle
 4) 00007FFA739FF72A (Qt5Core) : QTimer::timeout
 3) 00007FFA739F82DD (Qt5Core) : QObject::event
 2) 00007FFA668849A1 (Qt5Widgets) : QApplicationPrivate::notify_helper
 1) 00007FFA66883A23 (Qt5Widgets) : QApplication::notify
```

**Found in Branch**
development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
ba791dc178da125b1c07eec544d22c6719565b0f

**Additional context**
1. This is causing test failures in the Periodic builds: 

https://jenkins.build.o3de.org/blue/rest/organizations/jenkins/pipelines/O3DE_periodic-incremental-daily/branches/development/runs/271/nodes/1856/steps/1871/log/?start=0

https://jenkins.build.o3de.org/blue/rest/organizations/jenkins/pipelines/O3DE_periodic-incremental-daily/branches/development/runs/271/nodes/1260/steps/1279/log/?start=0

### PR DESCRIPTION
**Pull request by lumberyard-employee-dm**
The ScriptPropertyBoolean, ScriptPropertyNumber and ScriptPropertyGenericClass types are serialized with type name of "AzFramework::ScriptPropertyBoolean",
"AzFramework::ScriptPropertyNumber" and
"AzFramework::ScriptPropertyGenericClass" in *.prefab and *.scriptcanvas JSON serialized files.

fixes #14910

## How was this PR tested?

Validated a prefab file that contained an "AzFramework::ScriptPropertyBoolean" deserialized correctly.

### CODE REVIEW

File: Code/Framework/AzCore/AzCore/Script/ScriptProperty.cpp [hunk 1]
```diff
@@ -15,6 +15,49 @@
...
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyGenericClass, FunctionalScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyNumberArray, "ScriptPropertyNumberArray", "{76609A01-46CA-442E-8BA6-251D529886AF}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyNumberArray, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyBooleanArray, "ScriptPropertyBooleanArray", "{3A83958C-26C7-4A59-B6D7-A7805B0EC756}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyBooleanArray, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyStringArray, "ScriptPropertyStringArray", "{899993A5-D717-41BB-B89B-04A27952CA6D}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyStringArray, ScriptProperty);
```
Comment (mbalfour-amzn) [inline Code/Framework/AzCore/AzCore/Script/ScriptProperty.cpp:53]: I believe these should have AzFramework:: in front of their names.
https://github.com/o3de/o3de/blob/152bc0a1851d881fe735adf54ec93e1ad7875b11/Code/Framework/AzCore/AzCore/Script/ScriptProperty.h#L325

File: Code/Framework/AzCore/AzCore/Script/ScriptProperty.cpp [hunk 2]
```diff
@@ -15,6 +15,49 @@
...
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyGenericClassArray, "ScriptPropertyGenericClassArray", "{28E986DD-CF7C-404D-9BEE-EEE067180CD1}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyGenericClassArray, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyAsset, "ScriptPropertyAsset", "{4D4B7176-A6E1-4BB9-A7B0-5977EC724CCB}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyAsset, ScriptProperty);
```
Comment (mbalfour-amzn) [inline Code/Framework/AzCore/AzCore/Script/ScriptProperty.cpp:59]: I believe these should have AZ:: in front of their names.
https://github.com/o3de/o3de/blob/152bc0a1851d881fe735adf54ec93e1ad7875b11/Code/Framework/AzCore/AzCore/Script/ScriptProperty.h#L420

File: Code/Framework/AzCore/AzCore/Script/ScriptProperty.cpp [hunk 3]
```diff
@@ -15,6 +15,49 @@
...
+    // So make sure the old name is being used
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyBoolean, "AzFramework::ScriptPropertyBoolean", "{EA7335F8-5B9F-4744-B805-FEF9240451BD}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyBoolean, ScriptProperty);
+
+    // ScriptPropertyNumber is serialized to a prefab file using the type name of AzFramwork::ScriptPropertyNumber
+    // So make sure the old name is being used
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyNumber, "AzFramework::ScriptPropertyNumber", "{5BCDFDEB-A75D-4E83-BB74-C45299CB9826}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyNumber, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyString, "ScriptPropertyString", "{A0229C6D-B010-47E7-8985-EE220FC7BFAF}");
```
Comment (mbalfour-amzn) [inline Code/Framework/AzCore/AzCore/Script/ScriptProperty.cpp:38]: I believe this should have AzFramework:: in front of its name.
https://github.com/o3de/o3de/blob/152bc0a1851d881fe735adf54ec93e1ad7875b11/Code/Framework/AzCore/AzCore/Script/ScriptProperty.h#L220

Comment (lumberyard-employee-dm) [reply Code/Framework/AzCore/AzCore/Script/ScriptProperty.cpp:38]: The types other than ScriptPropertyBoolean, ScriptPropertyNumber and ScriptPropertyGenericClass, are serialized to any JSON files at the moment, but I can prefix their names with `AzFramework::` to be safe.

**Review by lawsonamzn — APPROVED** [review]

**Review by mbalfour-amzn — APPROVED** [review]

**Pull request merged**